engines/vcruise/textparser.cpp

```diff
@@ -0,0 +1,322 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "common/stream.h"
+#include "common/textconsole.h"
+
+#include "vcruise/textparser.h"
+
+namespace VCruise {
+
+TextParserState::TextParserState() : _lineNum(1), _col(1), _prevWasCR(false), _isParsingComment(false) {
+}
+
+TextParser::TextParser(Common::ReadStream *stream) : _stream(stream), _readBufferPos(0), _readBufferEnd(0), _returnBufferPos(kReturnBufferSize) {
+	memset(_readBuffer, 0, kReadBufferSize);
+	memset(_returnedBuffer, 0, kReturnBufferSize);
+}
+
+bool TextParser::readOneChar(char &outC, TextParserState &outState) {
+	if (_returnBufferPos == kReturnBufferSize) {
```

Comment:
I recommend using `wrapBufferedReadStream()` to achieve the same result. See `common/bufferedstream.h`

Reply:
Was this lost?

Reply:
It no longer buffers input, but it does buffer returned (requeued) characters, which is necessary because this was implemented with requeue instead of lookahead.